From: Ronan Le Martret <ronan@fridu.net>
Date: Tue, 22 Apr 2014 10:33:25 +0200
Subject: core: You can not put the cached result of use_smack fct,
 as we are not sure the "/sys" is mounted. So we should mount "sys"
 before "/proc"

https://bugs.freedesktop.org/show_bug.cgi?id=77646
(cherry picked from commit d1d8e5d49f3149d03ceb94d1d2f6c14e7abccb6f)
---
 src/core/mount-setup.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/core/mount-setup.c b/src/core/mount-setup.c
index ae8447c..991bfdf 100644
--- a/src/core/mount-setup.c
+++ b/src/core/mount-setup.c
@@ -67,10 +67,10 @@ typedef struct MountPoint {
 #define N_EARLY_MOUNT 5
 
 static const MountPoint mount_table[] = {
-        { "proc",       "/proc",                     "proc",       NULL, MS_NOSUID|MS_NOEXEC|MS_NODEV,
-          NULL,       MNT_FATAL|MNT_IN_CONTAINER },
         { "sysfs",      "/sys",                      "sysfs",      NULL, MS_NOSUID|MS_NOEXEC|MS_NODEV,
           NULL,       MNT_FATAL|MNT_IN_CONTAINER },
+        { "proc",       "/proc",                     "proc",       NULL, MS_NOSUID|MS_NOEXEC|MS_NODEV,
+          NULL,       MNT_FATAL|MNT_IN_CONTAINER },
         { "devtmpfs",   "/dev",                      "devtmpfs",   "mode=755", MS_NOSUID|MS_STRICTATIME,
           NULL,       MNT_FATAL|MNT_IN_CONTAINER },
         { "securityfs", "/sys/kernel/security",      "securityfs", NULL, MS_NOSUID|MS_NOEXEC|MS_NODEV,
